src/libsystemd/sd-netlink/test-netlink.c

```diff
@@ -156,27 +156,27 @@ static void test_route(void) {
 
         addr.s_addr = htonl(INADDR_LOOPBACK);
 
-        r = sd_netlink_message_append_in_addr(req, RTA_GATEWAY, &addr);
+        r = sd_netlink_message_append_in_addr(req1, RTA_GATEWAY, &addr);
         if (r < 0) {
                 log_error_errno(r, "Could not append RTA_GATEWAY attribute: %m");
                 return;
         }
 
-        r = sd_netlink_message_append_u32(req, RTA_OIF, index);
+        r = sd_netlink_message_append_u32(req1, RTA_OIF, index);
         if (r < 0) {
                 log_error_errno(r, "Could not append RTA_OIF attribute: %m");
                 return;
         }
 
-        assert_se(sd_netlink_message_rewind(req) >= 0);
+        assert_se(sd_netlink_message_rewind(req1) >= 0);
 
-        assert_se(sd_netlink_message_read_in_addr(req, RTA_GATEWAY, &addr_data) >= 0);
+        assert_se(sd_netlink_message_read_in_addr(req1, RTA_GATEWAY, &addr_data) >= 0);
         assert_se(addr_data.s_addr == addr.s_addr);
 
-        assert_se(sd_netlink_message_read_u32(req, RTA_OIF, &u32_data) >= 0);
+        assert_se(sd_netlink_message_read_u32(req1, RTA_OIF, &u32_data) >= 0);
         assert_se(u32_data == index);
 
-        assert_se((req = sd_netlink_message_unref(req)) == NULL);
+        assert_se((req2 = sd_netlink_message_unref(req1)) == NULL);
```

Comment:
This is wrong, because now you leave the req1 pointer set, but the message is already unref'ed. Hence, the cleanup function will unref it once more, causing memory corruption.